### PR TITLE
feat: Shared config support for UI

### PR DESCRIPTION
**Pull request by wainola**
## Description

- Implements support for shared-config on the UI
- Fee data settings are being loaded on demand when user select token

## Related Issue Or Context

Closes: #58 

## How Has This Been Tested? Testing details.

- Unit test to check the merge logic of the two configs

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [X] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation

## Checklist:

- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have ensured that all acceptance criteria (or expected behavior) from issue are met
- [ ] I have updated the documentation locally and in sygma-docs.
- [ ] I have added tests to cover my changes.
- [ ] I have ensured that all the checks are passing and green, I've signed the CLA bot

### CODE REVIEW

File: packages/sygma-ui/src/App.tsx [hunk 1]
```diff
@@ -44,9 +45,14 @@ const AppWrap: React.FC<{
...
-      const config = await getSygmaConfig();
-      if (config.error) {
-        setErrMessage(config.error.message ?? config.error.name);
+      // const config = await getSygmaConfig();
+      // const sharedConfig = await getSharedConfig()
+      const config = await configMerger();
+      if ((config as ConfigError).error) {
+        setErrMessage(
+          (config as ConfigError).error.message ??
+            (config as ConfigError).error.name
```
Comment (wainola) [inline packages/sygma-ui/src/App.tsx:54]: hey @enemycnt why are we using here property called `name`? In the config function if there is an error it returns an error with a message but not a name, so here I'm a little bit confused

File: packages/sygma-ui/src/types/sharedConfig.ts
Comment (wainola) [inline packages/sygma-ui/src/types/sharedConfig.ts:1]: Althoug I added this types for the `shared config`. some of them overlap with some of our types, mostly present in `BridgeConfig`. Will merge those here, so this is temporary

File: packages/sygma-ui/src/types/sygmaConfig.ts
Comment (wainola) [inline packages/sygma-ui/src/types/sygmaConfig.ts:1]: Althoug I added those type for the `shared-config` object that we are receiving, some of them overlap with our current types, mostly from some of the `BridgeConfig` definitions. I will merge them. This is temporary

File: packages/sygma-ui/src/App.tsx (hunk 1 above)
Comment (enemycnt) [reply packages/sygma-ui/src/App.tsx:54]: That is to catch connection error and server error from config server in a same time

**Review by enemycnt — APPROVED** [review]

**Pull request merged**